### PR TITLE
[FLINK-2734] [streaming] Fix ArrayKeySelector

### PR DESCRIPTION
**Pull request by StephanEwen**
The `ArrayKeySelector` is broken and returns wrong values in all cases except for `[0]` as a single only key position.

### CODE REVIEW

**Comment by StephanEwen** [comment]
The failed tests refer to an earlier version of the code.

The current version passe, see here: https://travis-ci.org/StephanEwen/incubator-flink/builds/81597319

File: flink-staging/flink-streaming/flink-streaming-core/src/test/java/org/apache/flink/streaming/util/keys/ArrayKeySelectorTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+			KeySelectorUtil.ArrayKeySelector<Object[]> twoFieldsSelector = new KeySelectorUtil.ArrayKeySelector<>(3, 0);
+			
+			assertEquals(new Tuple2<>("d", "a"), twoFieldsSelector.getKey(array1));
+			assertEquals(new Tuple2<>("y", "v"), twoFieldsSelector.getKey(array2));
+			
+		}
+		catch (Exception e) {
+			e.printStackTrace();
+			fail(e.getMessage());
+		}
```
Comment (uce) [inline flink-staging/flink-streaming/flink-streaming-core/src/test/java/org/apache/flink/streaming/util/keys/ArrayKeySelectorTest.java]: I don't think we need the try-catch block here.

File: flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/util/keys/KeySelectorUtil.java [hunk 2]
```diff
@@ -152,10 +119,9 @@ public ArrayKeySelector(int... fields) {
 
 		@Override
 		public Tuple getKey(IN value) throws Exception {
-			key = (Tuple) tupleClasses[fields.length - 1].newInstance();
+			Tuple key = Tuple.getTupleClass(fields.length).newInstance();
 			for (int i = 0; i < fields.length; i++) {
-				int pos = fields[i];
```
Comment (uce) [inline flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/util/keys/KeySelectorUtil.java]: This was the bug... Wow. That's super critical... I am wondering whether this should be part of 0.10 milestone 1

**Comment by uce** [comment]
+1 good catch and good test.

File: flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/util/keys/KeySelectorUtil.java (hunk 2 above)
Comment (mxm) [inline flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/util/keys/KeySelectorUtil.java]: How far are we with the release candidate? We might still be able to include it.

**Comment by StephanEwen** [comment]
Manually merged!

**Pull request closed without merging**